homeassistant/components/skybell/binary_sensor.py

```diff
@@ -12,90 +12,94 @@
-        key="device:sensor:button",
+BINARY_SENSOR_TYPES: tuple[BinarySensorEntityDescription, ...] = (
+    BinarySensorEntityDescription(
+        key="button",
         name="Button",
         device_class=BinarySensorDeviceClass.OCCUPANCY,
     ),
-    "motion": BinarySensorEntityDescription(
-        key="device:sensor:motion",
+    BinarySensorEntityDescription(
+        key="motion",
         name="Motion",
         device_class=BinarySensorDeviceClass.MOTION,
     ),
-}
-
+)
 
+# Deprecated in Home Assistant 2022.3
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Optional(
-            CONF_ENTITY_NAMESPACE, default=DEFAULT_ENTITY_NAMESPACE
-        ): cv.string,
+        vol.Optional(CONF_ENTITY_NAMESPACE, default=DOMAIN): cv.string,
         vol.Required(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
             cv.ensure_list, [vol.In(BINARY_SENSOR_TYPES)]
         ),
     }
 )
 
 
-def setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
-    """Set up the platform for a Skybell device."""
-    skybell = hass.data[SKYBELL_DOMAIN]
+    """Set up Skybell switch."""
+    skybell = hass.data[DOMAIN][entry.entry_id]
 
-    binary_sensors = [
-        SkybellBinarySensor(device, BINARY_SENSOR_TYPES[sensor_type])
-        for device in skybell.get_devices()
-        for sensor_type in config[CONF_MONITORED_CONDITIONS]
-    ]
+    sensors = []
+    for sensor in BINARY_SENSOR_TYPES:
+        for device in skybell[DATA_DEVICES]:
+            sensors.append(
+                SkybellBinarySensor(
+                    skybell[DATA_COORDINATOR],
+                    device,
+                    sensor,
+                    entry.entry_id,
+                )
+            )
 
-    add_entities(binary_sensors, True)
+    async_add_entities(sensors, True)
 
 
-class SkybellBinarySensor(SkybellDevice, BinarySensorEntity):
+class SkybellBinarySensor(SkybellEntity, BinarySensorEntity):
     """A binary sensor implementation for Skybell devices."""
 
     def __init__(
         self,
-        device,
+        coordinator: DataUpdateCoordinator,
+        device: SkybellDevice,
         description: BinarySensorEntityDescription,
-    ):
+        server_unique_id: str,
+    ) -> None:
         """Initialize a binary sensor for a Skybell device."""
-        super().__init__(device)
+        super().__init__(coordinator, device, server_unique_id)
         self.entity_description = description
-        self._attr_name = f"{self._device.name} {description.name}"
-        self._event: dict[Any, Any] = {}
+        self._attr_name = f"{device.name} {description.name}"
+        self._event: dict[str, str] = {}
+        self._attr_unique_id = f"{server_unique_id}/{description.key}"
```

Comment:
```suggestion
        self._attr_unique_id = f"{server_unique_id}_{description.key}"
```

Usually we separate unique ids with `-` and names with ` `